### PR TITLE
Implement delete cart item functionality

### PR DESCRIPTION
**Pull request by TienNguyenDinh**

### CODE REVIEW

File: javascript-practice/src/js/controllers/cart.controller.js [hunk 1]
```diff
@@ -92,7 +89,21 @@ export default class CartController {
...
   bindDeleteItemBtnEvent() {
+    const btnDeleteCartItemElements = document.querySelectorAll('.btn-delete-cart-item');
+
+    btnDeleteCartItemElements.forEach(element => {
+      element.addEventListener('click', async (e) => {
+        const id = e.target.dataset.cartItemId;
 
+        const data = await this.service.removeById(id);
+
+        if(data.success === false) {
```
Comment (hungnguyenhuu-agilityio) [inline javascript-practice/src/js/controllers/cart.controller.js]: shorthand pls

File: javascript-practice/src/js/controllers/product-detail.controller.js [hunk 2]
```diff
@@ -16,15 +17,50 @@ export default class ProductDetailController {
...
   bindAddCartBtnEvent() {
+    const addCartItemBtnElement = getElementById('btn-add-cart');
+
+    addCartItemBtnElement.addEventListener('click', async () => {
+      const id = getElementById('product-id').value;
+
+      const product = await this.productService.getById(id);
+      const { name, colors, price, imgUrl, amount: productAmount } = product;
+
+      const existingCartItem = await this.cartService.getByProductId(id);
```
Comment (huyencao-agilityio) [inline javascript-practice/src/js/controllers/product-detail.controller.js:39]: You can use `{ cartItemAmount } = await this.cartService.getByProductId(id);`

File: javascript-practice/src/js/controllers/product-detail.controller.js [hunk 3]
```diff
@@ -16,15 +17,50 @@ export default class ProductDetailController {
...
+      const { cartItemAmount } = existingCartItem;
+
+
+      if(productAmount <= 0) {
+        return showToastify('The product is running out of stock', 'toastify-danger');
+      }
+
+      // Needs to implement edit cart item first to expand this
+      // Update item amount if user already add it to cart
+      if(existingCartItem) {
```
Comment (huyencao-agilityio) [inline javascript-practice/src/js/controllers/product-detail.controller.js]: Should check condition before destructing obj `const { cartItemAmount } = existingCartItem;`

File: javascript-practice/src/js/controllers/product-detail.controller.js [hunk 4]
```diff
@@ -16,15 +17,50 @@ export default class ProductDetailController {
...
+      const id = getElementById('product-id').value;
+
+      const product = await this.productService.getById(id);
+      const { name, colors, price, imgUrl, amount: productAmount } = product;
+
+      const existingCartItem = await this.cartService.getByProductId(id);
+      const { cartItemAmount } = existingCartItem;
+
+
+      if(productAmount <= 0) {
```
Comment (huyencao-agilityio) [inline javascript-practice/src/js/controllers/product-detail.controller.js]: Code style

File: javascript-practice/src/js/controllers/product-detail.controller.js (hunk 3 above)
Comment (huyencao-agilityio) [inline javascript-practice/src/js/controllers/product-detail.controller.js]: Code style

File: javascript-practice/src/js/controllers/product-detail.controller.js (hunk 2 above)
Comment (huyencao-agilityio) [inline javascript-practice/src/js/controllers/product-detail.controller.js]: Naming convention

File: javascript-practice/src/js/services/cart.service.js [hunk 5]
```diff
@@ -35,9 +35,12 @@ export default class CartService {
 
   /**
    * Add a new cart item
+   * @param {Object} cartItem - the object contains the cart item
    */
-  async add() {
-
+  async add(cartItem) {
+    const endpoint = API_ROUTES.CART_ENDPOINT;
+    console.log(cartItem)
```
Comment (huyencao-agilityio) [inline javascript-practice/src/js/services/cart.service.js]: Remove console.log

File: javascript-practice/src/js/controllers/cart.controller.js [hunk 6]
```diff
@@ -92,7 +89,21 @@ export default class CartController {
    * Binds event to handle remove item from cart
    */
   bindDeleteItemBtnEvent() {
+    const btnDeleteCartItemElements = document.querySelectorAll('.btn-delete-cart-item');
+
+    btnDeleteCartItemElements.forEach(element => {
```
Comment (huyencao-agilityio) [inline javascript-practice/src/js/controllers/cart.controller.js]: Should use Event Delegation

File: javascript-practice/src/js/controllers/cart.controller.js [hunk 7]
```diff
@@ -92,7 +89,22 @@ export default class CartController {
...
+      element.addEventListener('click', async (e) => {
+        const id = e.target.dataset.cartItemId;
 
+        const data = await this.service.removeById(id);
+        const { success } = data;
+
+        if (success === false) {
```
Comment (huyencao-agilityio) [inline javascript-practice/src/js/controllers/cart.controller.js]: shorthand

File: javascript-practice/src/js/controllers/cart.controller.js [hunk 8]
```diff
@@ -92,7 +89,23 @@ export default class CartController {
    * Binds event to handle remove item from cart
    */
   bindDeleteItemBtnEvent() {
+    const cartListElement = document.querySelector('.shopping-cart');
+
+    cartListElement.addEventListener('click', async (e) => {
+      const btnDeleteCartItemElement = e.target.classList.contains('btn-delete-cart-item');
+
+      if(btnDeleteCartItemElement) {
```
Comment (huyencao-agilityio) [inline javascript-practice/src/js/controllers/cart.controller.js]: Code style

File: javascript-practice/src/js/controllers/cart.controller.js [hunk 9]
```diff
@@ -92,7 +89,23 @@ export default class CartController {
...
+        const id = e.target.dataset.cartItemId;
 
+        const { isSuccess } = await this.cartService.removeById(id);
+
+        if (!isSuccess) {
+          return;
```
Comment (huyencao-agilityio) [inline javascript-practice/src/js/controllers/cart.controller.js]: Should return a message here

**Pull request merged**